doc/mgmt/SONiC Management Framework Show Techsupport HLD.md

```diff
@@ -0,0 +1,198 @@
+## 2.2 Functional Description
+The set of items to be gathered for a given software release is defined by the development team. It is specified in a way that enables run-time access to the desired set of information items to be collected. The definition of the set of information items to be collected includes specification of the access function to be used for each item in the list to gather the information, format it as needed, and pack it into the output file. The location of the resulting output file is provided to the requesting client at the completion of command execution.
+
+# 3 Design
+## 3.1 Overview
+The "show techsupport" command causes invocation of an RPC sent from the management framework to a process in the host to cause collection of a list of flexibly defined sets of diagnostic information (information "items"). The collected list of items is stored in a compressed "tar" file with a unique name. The command output provides the location of the resulting compressed tar file.
+
+The "since" option can be used, if desired, to restrict the time scope for some of the information items (e.g. logs) to be collected. This options are passed to the host process for use during invocation of the applicable information gathering sub-functions.
+
+Help information and syntax details are provided if the command is executed with the "-h", "--help" or "-?" option.
+
+## 3.2 DB Changes
+N/A
+### 3.2.1 CONFIG DB
+### 3.2.2 APP DB
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+N/A
+### 3.3.1 Orchestration Agent
+### 3.3.2 Other Process
+The "show techsupport" feature requires RPC support in a process running within the host context. The host process handling the RPC is responsible for dispatching "show techsupport" requests from the management framework container to trigger allocation of an output file, gathering and packing of the required information into the output file, and sending a response to the management framework RPC agent to specify the name and path of the output file.
+
+## 3.4 SyncD
+N/A
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+The following Sonic Yang model is used for implementation of this feature:
+
+```module: sonic-show-techsupport
+
+  rpcs:
+    +---x sonic-show-techsupport-info
+       +---w input
+       |  +---w date?   yang:date-and-time
+       +--ro output
+          +--ro output-filename?   string
+
+
+
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
+N/A
+#### 3.6.2.2 Show Commands
+
+```
+Command syntax summary:
+
+show techsupport [since <DateTime\>]
```

Comment:
enclose this line in single backticks to make it monospaced. Like this:
`show techsupport [since <DateTime>]`

Reply:
Still need to fix this..